rfpremier/ah-django
# #164857229 Display article read time

**What does this PR do?**

Estimates the time it takes for a user to read an article

**Tasks to be completed**

- [x] Calculate article read time

**How can this be manually tested**

- Clone the repo and checkout the `ft/164857229-article-read-time `  branch

- Create and activate a virtual environment
```
$ virtualenv env
$ source env/bin/activate
```

- Install requirements
```
$ pip install -r requirements.txt
```

- Make migrations
```
$ python manage.py makemigrations
```

- Run migrations

```
$ python manage.py migrate
```

- Run the tests
```
python manage.py test
```
- Test via Postman 

  * Post an article
    `[POST] http://127.0.0.1:8000/api/articles/`
```
{
    "article": {
        "title": "This is the article title",
        "description": "This is the article description",
        "body": "Version control as the name suggests, is all about controlling the different versions of the application during its development. Git is probably the most common and easiest version control tool to use when developing an application.For the most part of development, I used(simulated) the Gitflow-Workflow. This workflow suggests that you create a develop branch off of the master branch and then create feature branches off the develop branch. The most important take way from this workflow is that the master branch should only contain ‘working’ code (code ready for release).",
        "image_url": "https://imageurl.com"
    }
}
```
   * Response
```
{
    "article": {
        "id": 6,
        "title": "This is the article title",
        "description": "This is the article description",
        "body": "Version control as the name suggests, is all about controlling the different versions of the application during its development. Git is probably the most common and easiest version control tool to use when developing an application.For the most part of development, I used(simulated) the Gitflow-Workflow. This workflow suggests that you create a develop branch off of the master branch and then create feature branches off the develop branch. The most important take way from this workflow is that the master branch should only contain ‘working’ code (code ready for release).",
        "image_url": "https://imageurl.com",
        "author": {
            "username": "bencyna",
            "bio": "",
            "image": ""
        },
        "created_at": "2019-04-16T08:37:14.795466Z",
        "updated_at": "2019-04-16T08:37:14.795539Z",
        "slug": "this-is-the-article-title-2",
        "avg_rating": 0,
        "rating_count": 0,
        "tags": [],
        "favorited": false,
        "favoritesCount": 0,
        "read_time": "1 min read"
    }
}
```
**Relevant Pivotal tracker stories** 
[#164857229](https://www.pivotaltracker.com/story/show/164857229)

### Screenshots

**Article Read Time**
![image](https://user-images.githubusercontent.com/11487123/56194825-69275380-603c-11e9-9452-22113f342578.png)
